### DISCUSSION TITLE
F-16 Startup Procedure In Unreal Engine

### DISCUSSION THREAD

**Discussion by Flight2IC** [JSBSim-Team/jsbsim#1137] [General]
Hi,
I am working with **f16.xml** as my FDM inside **unreal engine 5.3**, when i launch the game with "Start with engine running = true" everything works fine. But i want to start the engine using standard startup sequence which i am unable to achieve. I have used the following console commands but they didn't work either:

1. systems/electric/battery_switch = 1
2. systems/electric/apu_switch = 1
3. propulsion/engine[0]/starter_cmd = 1
4. fcs/throttle-cmd-norm = 0.1

In unreal engine blueprint setup i also tried setting engine "Running to true", "Starter to true" and "Generator Power to true". After trying different sets of instruction the N1, N2 or RPM stays at 0.

**Comment by bcoconni** [comment]
According to the script [scripts/f16_test.xml](https://github.com/JSBSim-Team/jsbsim/blob/0b99bf8b173fe21cd3a1481695ff199b5e4a4335/scripts/f16_test.xml), the procedure is the following:
1. Set `propulsion/starter_cmd` to 1
2. Wait for `propulsion/engine[0]/n2` to reach 15
3. Set `propulsion/cutoff_cmd` to 0

Extract from the file where the engine startup takes place:
https://github.com/JSBSim-Team/jsbsim/blob/0b99bf8b173fe21cd3a1481695ff199b5e4a4335/scripts/f16_test.xml#L9-L23

**Reply by Flight2IC** [reply]
thank you so much it was very helpful and the aircraft started immediately.
